interface-definitions/firewall.xml.in

```diff
@@ -205,6 +205,63 @@
               </leafNode>
             </children>
           </tagNode>
+          <tagNode name="external-list">
+            <properties>
+              <help>Firewall external lists</help>
+              <constraint>
+                <regex>[a-zA-Z0-9][\w\-\.]*</regex>
+              </constraint>
+            </properties>
+            <children>
+              <leafNode name="url">
```

Comment:
Please make use of the common URL CLI node. https://github.com/vyos/vyos-1x/blob/current/interface-definitions/include/url.xml.i

Reply:
Oh thank you, for some reason I couldn't find that one.  I'll fix that today

Reply:
Maybe url.xml.i needs to be updated to allow ftp, sftp, ftps, tftp, scp and then I use that? cause I can make that change if it seems okay, I can see that service-monitoring-telegraf and system-update-check use it, I haven't seen if they are capable to handle more then just http(s) though

Reply:
Oh and I need to be able to allow for local files, so using url.xml.i won't work here

Reply:
Removed ssh as that made no sense anyhow.

Maybe url.xml.i should be expanded to support ftp, sftp, ftps, tftp, scp and local file?  Then I could use it.

Reply:
I updated url.xml.i to allow scp:// and file:// so that local files and share files would be available and used url.xml.i in the firewall definition

Reply:
> I really wonder if a "user entered" cron job is really required. Why not make this refresh every 5,10 or 15 minutes? Is it bad to hardcode this?

I don't know where your comment went, but the reason to not have a static is very simple.

Most of these community lists will have limits and recommendations, I have some that are 24h limit, and other that have a 3h recommendation.

If you use a local file you'll likely want 5m so when you update the file it doesn't take forever to import.  This is why the design was to only update the lists requested on interval rather than all lists all the time.  If I went lowest common denominator I'd have to set it to 5m for the locals, even if I added a download interval so it only requests remote file so often with a lockfile for each, I'd still be processing 70k entries into nft every 5 minutes, that would not be very efficient.

This way only the much smaller local file is processed every 5m, where as larger files can be on 12h or 24h or what ever the list recommendation is.

That is the reason for this design.  Honestly I borrowed a lot of the concepts from OPNSense which I was testing before I decided on VyOS, it was one of the features I really liked and wanted to have.  So currently I have this running via bash scripts, but I figured others would likely benefit from it as well.